### PR TITLE
Split image building

### PR DESCRIPTION
**Pull request by YohannParis**
# Description

* Split all the images built with their own workflow
* ~Leverage `macos` to build arm version~ not possible until M2 available
* Remove the main `publish` file

All the images built, I deleted most of them, but kept the taskrunner until this is merged:
* https://github.com/orgs/DARPA-ASKEM/packages?tab=packages&q=task

### CODE REVIEW

File: .github/workflows/publish-client.yml [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+      - name: Setup Node 20.10.0 LTS
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20.10.0
+
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+
+      - name: Build image using Makefile
```
Comment (Nithos) [inline .github/workflows/publish-client.yml:60]: Why are you building here and then just below with the action? choose one method

File: .github/workflows/publish-client.yml [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+# yamllint disable-line rule:truthy
+on:
+  push:
+    paths: [ 'packages/client/**' ]
+    branches: [ 'main' ]
+    tags: [ '*' ]
+
+jobs:
+  image-name:
+    name: Create image name
```
Comment (Nithos) [inline .github/workflows/publish-client.yml:16]: you could use this for the tag generation, it gives you a few more perks but may be overkill here
[Metadata action](https://github.com/docker/metadata-action)

File: .github/workflows/publish-client.yml [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+
+      - name: Build image using Makefile
```
Comment (Nithos) [inline .github/workflows/publish-client.yml:94]: same as before, seems you are building twice

File: .github/workflows/publish-client.yml [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+      - arm64-emulation
+    steps:
+      - name: Login to registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Create, inspect and publish manifest
```
Comment (Nithos) [inline .github/workflows/publish-client.yml:157]: this looks correct to me

File: .github/workflows/publish-client.yml (hunk 1 above)
Comment (YohannParis) [reply .github/workflows/publish-client.yml:60]: It's just a poorly formatted name from the makefile as it doesn't build the image, but build the hmi-client (from source to distribution)

File: .github/workflows/publish-client.yml (hunk 2 above)
Comment (YohannParis) [reply .github/workflows/publish-client.yml:16]: After looking at it, yes I agree it is overkill.
Would be nice to have setup for Pantera though

**Review by dgauldie — APPROVED** [review]
ok

**Pull request merged**